PROCESSES.md

```diff
@@ -0,0 +1,147 @@
+The master information is in [apis.json](apis/apis.json) - the API
+catalog file. There's an entry for each API, containing:
+
+- The kind of API (grpc, rest, other)
+- The version number
+- A product title and home page for documentation
+- Dependencies
+- Additional test project dependencies
+- Target framework versions
+- Package description for NuGet
+- Tags for NuGet (in addition to default ones)
+
+The catalog is used to generate project files and also during the
+release process described below. Running the project generator is
+very simple: from the root directory, in a bash shell, run
+
+```bash
+./generateprojects.sh
+```
+
+The CI systems run this before building, to ensure that the project
+files are in a stable state.
+
+Generating the project files allows for broad changes (such as
+adding Source Link support) to be made very simply, just by changing
+the generator. Modifying every project file by hand simply doesn't
+scale.
+
+However, sometimes manual editing of project files is required. The
+project generator supports this by assuming it "owns":
+
+- The first `PropertyGroup` element (for general properties)
+- The first `ItemGroup` element (for dependencies)
+- The `ItemGroup` element for Source Link (with a label of "dotnet pack instructions")
+- The common import to only attempt to build desktop assemblies on Windows
+
+Any other elements are left as they are - so if you wish to add an
+`ItemGroup` such as for file grouping, just add it anywhere after
+the generated one, and it should do the right thing.
+
+Additionally, the project generator adds all projects under an API
+directory (and project references) to the solution. It will create
+project files from scratch as well - so when adding a new package
+from autogenerated API sources, the simplest approach is to copy the
+source files, modify the API catalog, and run the project generator.
+The project files and solution file will be generated and should
+immediately be usable.
+
+## Versioning
+
+All releasable packages follow [Semantic Versioning](http://semver.org).
+Non-releasable code (tests, tools, analyzers) are not versioned. The
+precise meaning of a breaking change (dictating version number
+increments) is out of the scope of this document. (Jon is writing a
+blog post about this topic and will link to it when it is published.)
+
+The version number in the API catalog (and therefore in project
+files) is one of:
+
+- The mostly recently released package version
+- The version that's *about* to be released (see below)
+- A prerelease with a suffix of 00 to indicate that the *next* release
+  should be a first prerelease, typically of a new minor version,
+  due to changes that have been merged. This is to avoid accidentally
+  including new features in a patch release. For example, having
+  released Google.Cloud.Storage.V1 version 2.0.0, new features were
+  merged into the `master` branch and the version was immediately
+  changed to 2.1.0-alpha00
```

Comment:
I would really prefer to keep the "last released version" approach. The main advantage is that it prevents accidental release of a new package - you can build all the packages and try to push them all to nuget, and they'll fail because they don't already exist.

I don't want us to automatically increment version numbers at the moment, as there are too many options to consider between "next alpha/beta", "promote from alpha/beta to beta/GA", "create patch from GA to GA", "move from GA to next minor version pre-release" or (for entirely-auto-generated packages) "move from GA to next minor GA".

The difference between the second and third cases is handled by `tagreleases.sh` - it knows never to tag anything with a `alpha/beta00` suffix. Clarified that in the doc.